### PR TITLE
daemon: fix print_pl only printing public zone peers

### PR DESCRIPTION
**Pull request by moneromooo-monero**

### CODE REVIEW

**Comment by radfish** [comment]
Hm, I'm sorry to say, but print_pl doesn't print the onion node from sync_info still...
```
$ monerod --config-file /etc/monerod.conf sync_info | grep -i onion
xmkwypann4ly64gh.onion:44469  a4d7b8b2c82733f0  normal            182       1867631  0 kB/s, 0 blocks / 0 MB queued
$ monerod --config-file /etc/monerod.conf print_pl | grep -i onion
$
```
```
$ monerod --version
Monero 'Boron Butterfly' (v0.14.1.0-633550972)
$ pacman -Qs monero
local/libmonero-wallet-git 0.14.1.0.r8344.gbd9641fe4-1
    Peer-to-peer anonymous digital currency (daemon, CLI wallet, and wallet API library)
local/monero-git 0.14.1.0.r8344.gbd9641fe4-1

```

Maybe my incremental build wasn't successful, but doubtful.

**Comment by moneromooo-monero** [comment]
What is your config so I can try ?

**Comment by moneromooo-monero** [comment]
Nevermind, I got the test case. I will fix.

**Comment by moneromooo-monero** [comment]
The daemon was just printing its own IPv4 specific thing rather than what the RPC was returning.

The peer ID also looks wonky, but it looks right on my work branch, I assume I fixed that and it'll display right after enough PRs get merged.

**Comment by radfish** [comment]
Confirmed. Works now. Thanks for fixing it!

File: src/p2p/net_node.inl [hunk 1]
```diff
@@ -1556,6 +1556,19 @@ namespace nodetool
   }
   //-----------------------------------------------------------------------------------
   template<class t_payload_net_handler>
+  void node_server<t_payload_net_handler>::get_peerlist(std::vector<std::vector<peerlist_entry>>& gray, std::vector<std::vector<peerlist_entry>>& white)
```
Comment (xiphon) [inline src/p2p/net_node.inl]: Propose to return merged gray and white lists, `zone.second.m_peerlist.get_peerlist` appends new entries to the lists, it doesn't clear them.

```c++
void node_server<t_payload_net_handler>::get_peerlist(std::vector<peerlist_entry>& gray, std::vector<peerlist_entry>& white)
{
    for (auto &zone: m_network_zones)
    {
      zone.second.m_peerlist.get_peerlist(gray, white);
    }
}
```

Comment (moneromooo-monero) [reply src/p2p/net_node.inl]: If it doesn't break any caller, sure.

**Comment by moneromooo-monero** [comment]
Rebased after the IPv6 patch. It would be nice if someone with IPv6 setup could check this works.

File: src/p2p/net_node.inl (hunk 1 above)
Comment (xiphon) [reply src/p2p/net_node.inl]: `node_server<t_payload_net_handler>::get_peerlist` is a new function introduced in this PR, changing it doesn't break any caller.

Implementing it the way i propose will omit a fair amount of changed lines in `core_rpc_server::on_get_peer_list` as well.

Comment (moneromooo-monero) [reply src/p2p/net_node.inl]: OK, I see what you mean now.

**Pull request merged**